design/deployment.md

```diff
@@ -0,0 +1,98 @@
+# A deployment job
+
+A "deployment" job is a collection of steps to be run sequentially against the environment. 
+We recommend specifying deployment steps in a deployment job.  
+
+## Scenarios
+
+- Run a set of steps against an environment. Record deployment history and status of the deployments.
+- Perform safe deployments. In other words, define how your application is rolled-out.
+
+  
+### Schema
+
+```yaml
+jobs:
+- deployment: string  # name of the deployment job, A-Z, a-z, 0-9, and underscore
+  displayName: string  # friendly name to display in the UI
+  dependsOn: string | [ string ]
+  environment: string # target environment name and optionally a resource-name to record the deployment history; format: <environment-name>.<resource-name>
+  condition: string
+  continueOnError: boolean  # 'true' if future jobs should run even if this job fails; defaults to 'false'
+  pool: pool # see pool schema
+  timeoutInMinutes: number # how long to run the job before automatically cancelling
+  cancelTimeoutInMinutes: number # how much time to give 'run always even if cancelled tasks' before killing them
+  variables: { string: string } | [ variable | variableReference ] 
+  strategy: 
+    runOnce: # default runOnce strategy, useful for running the steps sequentially once against the environment.
+      deploy:
+        displayName: string # friendly name to display in the UI
```

Comment:
Can we have filters (or tags) added?

Reply:
Updated environment schema including `tags` filter for the resources in the environments.md file. URL property is yet to be finalized